kernel/packages/unity-interface/BrowserInterface.ts

```diff
@@ -377,6 +378,45 @@ export class BrowserInterface {
     globalThis.globalStore.dispatch(updateFriendship(action, userId.toLowerCase(), false))
   }
 
+  public SearchENSOwner(data: { name: string; maxResults: number }) {
+    let profilesPromise
+
+    if (/^0x[a-fA-F0-9]{40}$/.test(data.name)) {
+      profilesPromise = ProfileAsPromise(data.name, undefined, ProfileType.DEPLOYED).then((profile) => {
+        return [profile]
+      })
+    } else {
+      profilesPromise = new Promise((resolve, reject) => {
+        (async () => {
+          try {
+            const net = await getAppNetwork()
+            const owners = await fetchENSOwnersContains(ethereumConfigurations[net].names, data.name, data.maxResults)
+            const profiles: Profile[] = []
+            for (let userId of owners) {
+              await ProfileAsPromise(userId, undefined, ProfileType.DEPLOYED).then((profile) => {
+                profiles.push(profile)
+              })
+            }
+            resolve(profiles)
+          } catch (error) {
+            reject(error)
+          }
+        })()
+        .catch(error => defaultLogger.error(error))
+      })
```

Comment:
What about something like this? I think it would be a little bit more clear

```
let userIds: Promise<string[]> 

if (/^0x[a-fA-F0-9]{40}$/.test(data.name)) {
  addresses = Promise.resolve([data.name])
} else {
  addresses = getAppNetwork()
    .then(net => fetchENSOwnersContains(ethereumConfigurations[net].names, data.name, data.maxResults))
}

const profilesPromise = userIds
  .then(userIds => userIds.map(userId => ProfileAsPromise(userId, undefined, ProfileType.DEPLOYED)))
  .catch(error => defaultLogger.error(error))
```

Disclaimer: haven't tested it

Reply:
Thx for this suggestion! I implemented with little changes